internal/handlers/user.go

```diff
@@ -22,14 +22,24 @@ const AuthUserName string = "_nexodus.UserName"
 
 var noUUID = uuid.UUID{}
 
+// CacheExp Zero expiration means the key has no expiration time.
+const CacheExp time.Duration = 0
+
 func (api *API) CreateUserIfNotExists() gin.HandlerFunc {
 	return func(c *gin.Context) {
 		id := c.GetString(gin.AuthUserKey)
 		username := c.GetString(AuthUserName)
-		_, err := api.createUserIfNotExists(c.Request.Context(), id, username)
+		cachedUsername, err := api.redis.Get(c.Request.Context(), id).Result()
 		if err != nil {
-			_ = c.AbortWithError(http.StatusInternalServerError, err)
-			return
+			api.logger.Infof("failed to find user in the cache:%s", err)
```

Comment:
A cache miss will be a common scenario.  If the error is because of a cache miss, I would recommend debug logging level, otherwise if the error is due to a redis system error, a warn level would be better.